### PR TITLE
Add logic to only show group members on free-for-all factions

### PR DESCRIPTION
**Pull request by paul2kandrews**
This branch allows Player Map Markers to render the positions of group members to each other on a faction that is not friendly to itself. It also forces the position of the local player to always be shown.

Closes issue #7

### CODE REVIEW

File: addons/PlayerMapMarkers/scripts/Game/FPM/Groups/SCR_GroupsManagerComponent.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
+modded class SCR_GroupsManagerComponent
```
Comment (Kexanone) [inline addons/PlayerMapMarkers/scripts/Game/FPM/Groups/SCR_GroupsManagerComponent.c:1]:
```suggestion
//------------------------------------------------------------------------------------------------
modded class SCR_GroupsManagerComponent : SCR_BaseGameModeComponent
```

File: addons/PlayerMapMarkers/scripts/Game/FPM/Groups/SCR_GroupsManagerComponent.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		super.OnGroupPlayerAdded(group, playerID);
+		
+		if (IsProxy())
+			return;
+		
+		UpdateMarkerStreamRulesForAllGroupMembers(group);
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	private void UpdateMarkerStreamRulesForAllGroupMembers(SCR_AIGroup group)
```
Comment (Kexanone) [inline addons/PlayerMapMarkers/scripts/Game/FPM/Groups/SCR_GroupsManagerComponent.c:28]:
```suggestion
	protected void UpdateMarkerStreamRulesForAllGroupMembers(SCR_AIGroup group)
```

File: addons/PlayerMapMarkers/scripts/Game/FPM/Map/Markers/SCR_MapMarkerManagerComponent.c [hunk 3]
```diff
@@ -2,21 +2,71 @@
...
 		{
 			// FPM: Added exception handling
 			if (!dynamicMarker)
 				continue;
 			
-			Faction markerFaction = dynamicMarker.GetFaction();
+			// Cast to FPM so we can get the player this marker is associated with
+			FPM_MapMarkerPlayer fpmMarker = FPM_MapMarkerPlayer.Cast(dynamicMarker);
+			if (!fpmMarker)
+				continue;
```
Comment (Kexanone) [inline addons/PlayerMapMarkers/scripts/Game/FPM/Map/Markers/SCR_MapMarkerManagerComponent.c:17]: This is no good. You basically break any other type of dynamic markers that way.

**Review by Kexanone — CHANGES_REQUESTED** [review]

File: addons/PlayerMapMarkers/scripts/Game/FPM/Map/Markers/SCR_MapMarkerManagerComponent.c (hunk 3 above)
Comment (paul2kandrews) [reply addons/PlayerMapMarkers/scripts/Game/FPM/Map/Markers/SCR_MapMarkerManagerComponent.c]: Addressed in fb90c8a

**Pull request closed without merging**